File: Hot100/94. 二叉树的中序遍历.go
package main

// 94. 二叉树的中序遍历

// 给定一个二叉树的根节点 root ，返回它的 中序 遍历。

func main() {

}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	result := make([]int, 0)
	inorder(root, &result)
	return result
}

func inorder(node *TreeNode, result *[]int) {
	if node == nil {
		return
	}

	inorder(node.Left, result)
	*result = append(*result, node.Val)
	inorder(node.Right, result)
}

File: C++/94.二叉树的中序遍历.cpp

// 94. 二叉树的中序遍历
//
// 给定一个二叉树的根节点 root ，返回 它的 中序 遍历 。

#include <vector>

#include "header.h"

class Solution {
public:
    std::vector<int> inorderTraversal(TreeNode* root) {
        std::vector<int> res;
        inorder(root, res);
        return res;
    }

    void inorder(TreeNode* root, std::vector<int>& res) {
        if (root == nullptr) {
            return;
        }

        inorder(root->left, res);
        res.emplace_back(root->val);
        inorder(root->right, res);
    }
};
